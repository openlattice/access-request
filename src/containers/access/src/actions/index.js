/*
 * @flow
 */

import { newRequestSequence } from 'redux-reqseq';
import type { Map } from 'immutable';
import type { RequestSequence } from 'redux-reqseq';

const GET_FORMS :'GET_FORMS' = 'GET_FORMS';
const getForms :RequestSequence = newRequestSequence(GET_FORMS);

const GET_ALL_ACCESS_REQUESTS :'GET_ALL_ACCESS_REQUESTS' = 'GET_ALL_ACCESS_REQUESTS';
const getAllAccessRequests :RequestSequence = newRequestSequence(GET_ALL_ACCESS_REQUESTS);

const GET_ACCESS_REQUEST :'GET_ACCESS_REQUEST' = 'GET_ACCESS_REQUEST';
const getAccessRequest :RequestSequence = newRequestSequence(GET_ACCESS_REQUEST);

const SUBMIT_ACCESS_REQUEST :'SUBMIT_ACCESS_REQUEST' = 'SUBMIT_ACCESS_REQUEST';
const submitAccessRequest :RequestSequence = newRequestSequence(SUBMIT_ACCESS_REQUEST);

const SELECT_ACCESS_REQUEST :'SELECT_ACCESS_REQUEST' = 'SELECT_ACCESS_REQUEST';
const selectAccessRequest = (value :Map) => ({
  type: SELECT_ACCESS_REQUEST,
  value
});

const CLEAR_ACCESS_REQUEST :'CLEAR_ACCESS_REQUEST' = 'CLEAR_ACCESS_REQUEST';
const clearAccessRequest = () => ({
  type: CLEAR_ACCESS_REQUEST
});

const UPDATE_ACCESS_REQUEST :'UPDATE_ACCESS_REQUEST' = 'UPDATE_ACCESS_REQUEST';
const updateAccessRequest :RequestSequence = newRequestSequence(UPDATE_ACCESS_REQUEST);

const GET_ATTACHMENTS :'GET_ATTACHMENTS' = 'GET_ATTACHMENTS';
const getAttachments :RequestSequence = newRequestSequence(GET_ATTACHMENTS);

const UPLOAD_ATTACHMENTS :'UPLOAD_ATTACHMENTS' = 'UPLOAD_ATTACHMENTS';
const uploadAttachments :RequestSequence = newRequestSequence(UPLOAD_ATTACHMENTS);

const DELETE_ATTACHMENTS :'DELETE_ATTACHMENTS' = 'DELETE_ATTACHMENTS';
const deleteAttachments :RequestSequence = newRequestSequence(DELETE_ATTACHMENTS);

const SAVE_NEW_FORM_TEMPLATE :'SAVE_NEW_FORM_TEMPLATE' = 'SAVE_NEW_FORM_TEMPLATE';
const saveNewFormTemplate :RequestSequence = newRequestSequence(SAVE_NEW_FORM_TEMPLATE);

const UPDATE_ATTACHMENT_TAG :'UPDATE_ATTACHMENT_TAG' = 'UPDATE_ATTACHMENT_TAG';
const updateAttachmentTag :RequestSequence = newRequestSequence(UPDATE_ATTACHMENT_TAG);

export {
  CLEAR_ACCESS_REQUEST,
  DELETE_ATTACHMENTS,
  GET_ACCESS_REQUEST,
  GET_ALL_ACCESS_REQUESTS,
  GET_ATTACHMENTS,
  GET_FORMS,
  SAVE_NEW_FORM_TEMPLATE,
  SELECT_ACCESS_REQUEST,
  SUBMIT_ACCESS_REQUEST,
  UPDATE_ACCESS_REQUEST,
  UPDATE_ATTACHMENT_TAG,
  UPLOAD_ATTACHMENTS,
  clearAccessRequest,
  deleteAttachments,
  getAccessRequest,
  getAllAccessRequests,
  getAttachments,
  getForms,
  saveNewFormTemplate,
  selectAccessRequest,
  submitAccessRequest,
  updateAccessRequest,
  updateAttachmentTag,
  uploadAttachments,
};
