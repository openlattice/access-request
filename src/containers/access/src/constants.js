// @flow
const CONSENT_FORM :'Consent Form' = 'Consent Form';
const CPHS_REVIEW :'CPHS Review' = 'CPHS Review';
const DATA_RELEASE_LETTER :'Data Release Letter' = 'Data Release Letter';
const DATA_SECURITY_LETTER :'Data Security Letter' = 'Data Security Letter';
const DATA_USE_AGREEMENT :'Data Use Agreement' = 'Data Use Agreement';
const DEPARTMENT_SUPPORT_LETTER :'Department Support Letter' = 'Department Support Letter';
const IRB_APPROVAL :'IRB Approval' = 'IRB Approval';
const PARTICIPANTS_BOR :"Participant's Bill of Rights" = "Participant's Bill of Rights";

const TAG_OPTIONS = [
  { label: CONSENT_FORM, value: CONSENT_FORM },
  { label: CPHS_REVIEW, value: CPHS_REVIEW },
  { label: DATA_RELEASE_LETTER, value: DATA_RELEASE_LETTER },
  { label: DATA_SECURITY_LETTER, value: DATA_SECURITY_LETTER },
  { label: DATA_USE_AGREEMENT, value: DATA_USE_AGREEMENT },
  { label: DEPARTMENT_SUPPORT_LETTER, value: DEPARTMENT_SUPPORT_LETTER },
  { label: IRB_APPROVAL, value: IRB_APPROVAL },
  { label: PARTICIPANTS_BOR, value: PARTICIPANTS_BOR },
];

export {
  CONSENT_FORM,
  CPHS_REVIEW,
  DATA_RELEASE_LETTER,
  DATA_SECURITY_LETTER,
  DATA_USE_AGREEMENT,
  DEPARTMENT_SUPPORT_LETTER,
  IRB_APPROVAL,
  PARTICIPANTS_BOR,
  TAG_OPTIONS,
};
