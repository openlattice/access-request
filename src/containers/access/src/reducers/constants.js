export const ACCESS_REQUEST = 'accessRequest';
export const ATTACHMENTS = 'attachments';
export const FORMS = 'forms';
