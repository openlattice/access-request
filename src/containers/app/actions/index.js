/*
 * @flow
 */

import { newRequestSequence } from 'redux-reqseq';
import type { RequestSequence } from 'redux-reqseq';

const INITIALIZE_APPLICATION :'INITIALIZE_APPLICATION' = 'INITIALIZE_APPLICATION';
const initializeApplication :RequestSequence = newRequestSequence(INITIALIZE_APPLICATION);

const SET_ROOT :'SET_ROOT' = 'SET_ROOT';
const setRoot = (value :string) => ({
  type: SET_ROOT,
  value
});

export {
  INITIALIZE_APPLICATION,
  initializeApplication,
  SET_ROOT,
  setRoot,
};
