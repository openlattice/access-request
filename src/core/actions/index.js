// @flow

import { newRequestSequence } from 'redux-reqseq';
import type { RequestSequence } from 'redux-reqseq';

const DELETE_BULK_ENTITIES :'DELETE_BULK_ENTITIES' = 'DELETE_BULK_ENTITIES';
const deleteBulkEntities :RequestSequence = newRequestSequence(DELETE_BULK_ENTITIES);

export {
  DELETE_BULK_ENTITIES,
  deleteBulkEntities,
};
