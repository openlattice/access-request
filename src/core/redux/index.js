import { createContext } from 'react';

import {
  createDispatchHook,
  createSelectorHook,
  createStoreHook
} from 'react-redux';

import initializeReduxStore from './ReduxStore';

import { routerHistory } from '../router';

export const moduleStore = initializeReduxStore(routerHistory);
export const moduleContext = createContext(null);

// Export your custom hooks if you wish to use them in other files.
export const useStore = createStoreHook(moduleContext);
export const useDispatch = createDispatchHook(moduleContext);
export const useSelector = createSelectorHook(moduleContext);
