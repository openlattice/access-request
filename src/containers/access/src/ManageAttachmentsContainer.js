// @flow
import { useEffect } from 'react';

// $FlowFixMe
import { List } from 'lattice-ui-kit';
import { DataUtils } from 'lattice-utils';
import type { UUID } from 'lattice';

import AttachmentItem from './AttachmentItem';
import { deleteAttachments, getAttachments } from './actions';
import { ATTACHMENTS } from './reducers/constants';

import { useDispatch, useSelector } from '../../../core/redux';
import { ACCESS } from '../../../core/redux/constants';

const { getEntityKeyId } = DataUtils;
type Props = {
  accessRequestId :UUID;
};

const ManageAttachmentsContainer = ({ accessRequestId } :Props) => {
  const attachments = useSelector((state) => state.getIn([ACCESS, ATTACHMENTS]));
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getAttachments(accessRequestId));
  }, [accessRequestId, dispatch]);

  const onDelete = (fileId) => {
    dispatch(deleteAttachments([fileId]));
  };

  return (
    <List>
      {
        attachments.valueSeq().map((file, index) => {
          const fileId = getEntityKeyId(file);
          const divider = index !== attachments.size - 1;

          return (
            <AttachmentItem
                divider={divider}
                file={file}
                key={fileId}
                onDelete={onDelete} />
          );
        })
      }
    </List>
  );
};

export default ManageAttachmentsContainer;
