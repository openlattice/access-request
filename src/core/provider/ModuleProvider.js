// @flow

import { ConnectedRouter } from 'connected-react-router/immutable';
import { Provider } from 'react-redux';

import { moduleContext, moduleStore } from '../redux';
import { routerHistory } from '../router';

type Props = {
  children :any;
}

export default function ModuleProvider({ children } :Props) {
  return (
    <Provider context={moduleContext} store={moduleStore}>
      <ConnectedRouter history={routerHistory} context={moduleContext}>
        {children}
      </ConnectedRouter>
    </Provider>
  );
}
