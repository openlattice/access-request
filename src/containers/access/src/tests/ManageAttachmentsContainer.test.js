import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import { Map } from 'immutable';
import { List } from 'lattice-ui-kit';
import { NIL } from 'uuid';

import { MOCK_ATTACHMENTS } from './constants';

import ManageAttachmentsContainer from '../ManageAttachmentsContainer';
import ModuleProvider from '../../../../core/provider/ModuleProvider';
import { selectAttachments } from '../../../../core/redux/selectors';
import { DELETE_ATTACHMENTS, GET_ATTACHMENTS } from '../actions';

const mockDispatch = jest.fn();
const mockUseDispatch = jest.fn();
mockUseDispatch.mockReturnValue(mockDispatch);

jest.mock('../../../../core/redux', () => {
  const redux = jest.requireActual('../../../../core/redux');
  return {
    ...redux,
    useDispatch: () => mockUseDispatch(),
  };
});

jest.mock('../../../../core/redux/selectors', () => ({
  selectAttachments: jest.fn()
}));

describe('ManageAttachmentsContainer', () => {
  beforeEach(() => {
    selectAttachments.mockImplementation(() => () => Map());
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('match snapshot', () => {
    const wrapper = mount(
      <ModuleProvider>
        <ManageAttachmentsContainer accessRequestId={NIL} />
      </ModuleProvider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('return empty list by default', () => {
    const wrapper = mount(
      <ModuleProvider>
        <ManageAttachmentsContainer accessRequestId={NIL} />
      </ModuleProvider>
    );

    expect(wrapper.find('list').children()).toHaveLength(0);
  });

  test('return list of attachments', () => {
    selectAttachments.mockImplementation(() => () => MOCK_ATTACHMENTS);
    const wrapper = mount(
      <ModuleProvider>
        <ManageAttachmentsContainer accessRequestId={NIL} />
      </ModuleProvider>
    );

    expect(wrapper.find(List).children()).toHaveLength(1);
  });

  test('dispatch GET_ATTACHMENTS with accessRequestId on mount', () => {
    const wrapper = mount(
      <ModuleProvider>
        <ManageAttachmentsContainer accessRequestId={NIL} />
      </ModuleProvider>
    );

    const attachmentsBodyWrapper = wrapper.find(ManageAttachmentsContainer);
    expect(attachmentsBodyWrapper.prop('accessRequestId')).toEqual(NIL);
    expect(mockDispatch).toHaveBeenCalledTimes(1);
    expect(mockDispatch.mock.calls[0][0]).toHaveProperty('id');
    expect(mockDispatch.mock.calls[0][0]).toHaveProperty('type', GET_ATTACHMENTS);
    expect(mockDispatch.mock.calls[0][0]).toHaveProperty('value', NIL);
  });

  test('dispatch DELETE_ATTACHMENTS when onDelete is called', () => {
    selectAttachments.mockImplementation(() => () => MOCK_ATTACHMENTS);
    const wrapper = mount(
      <ModuleProvider>
        <ManageAttachmentsContainer accessRequestId={NIL} />
      </ModuleProvider>
    );

    const deleteButtonWrapper = wrapper.find('button[aria-label="Delete"]');
    expect(deleteButtonWrapper).toHaveLength(1);
    deleteButtonWrapper.simulate('click');
    expect(mockDispatch.mock.calls[1][0]).toHaveProperty('id');
    expect(mockDispatch.mock.calls[1][0]).toHaveProperty('type', DELETE_ATTACHMENTS);
    expect(mockDispatch.mock.calls[1][0]).toHaveProperty('value', [NIL]);
  });

});
