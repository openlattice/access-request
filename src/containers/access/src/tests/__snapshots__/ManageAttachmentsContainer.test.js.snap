// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageAttachmentsContainer match snapshot 1`] = `
<ModuleProvider>
  <Provider
    context={
      Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": Object {},
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      }
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(ConnectedRouterWithContext)
      context={
        Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": Object {},
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        }
      }
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <ConnectedRouterWithContext
        context={
          Object {
            "$$typeof": Symbol(react.context),
            "Consumer": Object {
              "$$typeof": Symbol(react.context),
              "_calculateChangedBits": null,
              "_context": [Circular],
            },
            "Provider": Object {
              "$$typeof": Symbol(react.provider),
              "_context": [Circular],
            },
            "_calculateChangedBits": null,
            "_currentRenderer": Object {},
            "_currentRenderer2": null,
            "_currentValue": null,
            "_currentValue2": null,
            "_threadCount": 0,
          }
        }
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        onLocationChanged={[Function]}
      >
        <ConnectedRouter
          context={
            Object {
              "$$typeof": Symbol(react.context),
              "Consumer": Object {
                "$$typeof": Symbol(react.context),
                "_calculateChangedBits": null,
                "_context": [Circular],
              },
              "Provider": Object {
                "$$typeof": Symbol(react.provider),
                "_context": [Circular],
              },
              "_calculateChangedBits": null,
              "_currentRenderer": Object {},
              "_currentRenderer2": null,
              "_currentValue": null,
              "_currentValue2": null,
              "_threadCount": 0,
            }
          }
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          onLocationChanged={[Function]}
          store={
            Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            }
          }
        >
          <Router
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
          >
            <ManageAttachmentsContainer
              accessRequestId="00000000-0000-0000-0000-000000000000"
            >
              <ForwardRef(WithStyles)>
                <ForwardRef(List)
                  classes={
                    Object {
                      "dense": "MuiList-dense",
                      "padding": "MuiList-padding",
                      "root": "MuiList-root",
                      "subheader": "MuiList-subheader",
                    }
                  }
                >
                  <ul
                    className="MuiList-root MuiList-padding"
                  />
                </ForwardRef(List)>
              </ForwardRef(WithStyles)>
            </ManageAttachmentsContainer>
          </Router>
        </ConnectedRouter>
      </ConnectedRouterWithContext>
    </Connect(ConnectedRouterWithContext)>
  </Provider>
</ModuleProvider>
`;
